### PR TITLE
fix(scanv4): Major Scanner V4 Image Unification

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

Related to

* https://github.com/stackrox/stackrox/pull/10153
* [this Slack thread](https://redhaty-internal.slack.com/archives/C033Z8KMZAM/p1709119914557639)

I understand that one of the very early design decisions which were made for the Helm chart integration turns out to be problematic.

I know that this is *very* late right before the code freeze, but I wanted to give it a last-minute try to restructure what we have.

I have tagged so many reviewers due to the importance here to not break things given that we have very little soak time.

What this PR essentially does is removing the `scannerV4.{indexer,matcher}.image` configuration settings, as a replacement it introduces `scannerV4.image`. I made the attempt to keep it minimally invasive (for example, for the Helm charts I have only changed the outer interface).

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: operator/pkg/central/values/translation/image_overrides.go [hunk 1]
```diff
@@ -4,12 +4,11 @@ import "github.com/stackrox/rox/operator/pkg/images"
...
-		images.ScannerDB:        "scanner.dbImage.fullRef",
-		images.ScannerV4DB:      "scannerV4.db.image.fullRef",
-		images.ScannerV4Indexer: "scannerV4.indexer.image.fullRef",
-		images.ScannerV4Matcher: "scannerV4.matcher.image.fullRef",
+		images.Main:        "central.image.fullRef",
+		images.CentralDB:   "central.db.image.fullRef",
+		images.Scanner:     "scanner.image.fullRef",
+		images.ScannerDB:   "scanner.dbImage.fullRef",
+		images.ScannerV4DB: "scannerV4.db.image.fullRef",
+		images.ScannerV4:   "scannerV4.image.fullRef",
```
Comment (RTann) [inline operator/pkg/central/values/translation/image_overrides.go:12]: nit: reorder for consistency

File: operator/pkg/images/env.go [hunk 2]
```diff
@@ -4,15 +4,14 @@ import "github.com/stackrox/rox/pkg/env"
...
+	Main          = env.RegisterSetting("RELATED_IMAGE_MAIN")
+	CentralDB     = env.RegisterSetting("RELATED_IMAGE_CENTRAL_DB")
+	Scanner       = env.RegisterSetting("RELATED_IMAGE_SCANNER")
+	ScannerSlim   = env.RegisterSetting("RELATED_IMAGE_SCANNER_SLIM")
+	ScannerDB     = env.RegisterSetting("RELATED_IMAGE_SCANNER_DB")
+	ScannerSlimDB = env.RegisterSetting("RELATED_IMAGE_SCANNER_DB_SLIM")
+	CollectorSlim = env.RegisterSetting("RELATED_IMAGE_COLLECTOR_SLIM")
+	CollectorFull = env.RegisterSetting("RELATED_IMAGE_COLLECTOR_FULL")
+	ScannerV4DB   = env.RegisterSetting("RELATED_IMAGE_SCANNER_V4_DB")
+	ScannerV4     = env.RegisterSetting("RELATED_IMAGE_SCANNER_V4")
```
Comment (RTann) [inline operator/pkg/images/env.go:16]: nit: reorder

File: operator/pkg/securedcluster/values/translation/image_overrides.go [hunk 3]
```diff
@@ -4,12 +4,12 @@ import "github.com/stackrox/rox/operator/pkg/images"
...
-		images.ScannerSlimDB:    "image.scannerDb.fullRef",
-		images.ScannerV4DB:      "image.scannerV4DB.fullRef",
-		images.ScannerV4Indexer: "image.scannerV4.fullRef",
+		images.Main:          "image.main.fullRef",
+		images.CollectorSlim: "image.collector.slim.fullRef",
+		images.CollectorFull: "image.collector.full.fullRef",
+		images.ScannerSlim:   "image.scanner.fullRef",
+		images.ScannerSlimDB: "image.scannerDb.fullRef",
+		images.ScannerV4DB:   "image.scannerV4DB.fullRef",
+		images.ScannerV4:     "image.scannerV4.fullRef",
```
Comment (RTann) [inline operator/pkg/securedcluster/values/translation/image_overrides.go:13]: nit: reorder

**Review by RTann — APPROVED** [review]
definitely get a review from someone who is more familiar with helm/operator, but LGTM

**Review by ludydoo — APPROVED** [review]

**Review by johannes94 — APPROVED** [review]

File: operator/pkg/central/values/translation/image_overrides.go (hunk 1 above)
Comment (mclasmeier) [reply operator/pkg/central/values/translation/image_overrides.go:12]: sorry, have only seen that now. I suggest to proceed.

File: operator/pkg/securedcluster/values/translation/image_overrides.go (hunk 3 above)
Comment (mclasmeier) [reply operator/pkg/securedcluster/values/translation/image_overrides.go:13]: seen this one too late, sorry. happy to clean this up later.

**Pull request merged**